demosplan/DemosPlanCoreBundle/Controller/Statement/DemosPlanStatementController.php

```diff
@@ -2350,9 +2357,45 @@ public function readOnlyStatementListAction(
     #[Route(name: 'DemosPlan_statement_import', methods: ['POST'], path: '/verfahren/{procedureId}/stellungnahmen/import', options: ['expose' => true])]
     public function importStatementsAction(
         FileService $fileService,
-        PermissionsInterface $permissions,
         ProcedureService $procedureService,
-        XlsxStatementImport $importer,
+        XlsxStatementImporterFactory $importerFactory,
+        ExcelImporter $excelImporter,
+        string $procedureId,
+        Request $request
+    ): Response {
+        $requestPost = $request->request->all();
+        $procedure = $procedureService->getProcedure($procedureId);
+
+        if (null === $procedure) {
+            throw ProcedureNotFoundException::createFromId($procedureId);
+        }
+
+        // recreate uploaded array
+        $uploads = explode(',', (string) $requestPost['uploadedFiles']);
+        $files = array_map([$fileService, 'getFileInfo'], $uploads);
+        $importer = $importerFactory->createXlsxStatementImporter($excelImporter);
+
+        return $this->importStatements($files, $procedureId, $importer);
```

Comment:
This entire code part is duplicated. Is the reason the different permission or is it necessary to keep some kind of loose coupling for the addon system? I'm a bit confused.

Reply:
minor only, could be better, but this isn't a blocker